common/kvstore/batch.go

```diff
@@ -0,0 +1,20 @@
+package kvstore
+
+// Batch is a collection of operations that can be applied atomically to a TableStore.
+type Batch[T any] interface {
```

Comment:
Not sure this needs template, Batch may just work with table key

Reply:
The issue is this interface intends to be used for `TableStore` (like it comments out), but it's now got used to batch operations for `Store` as well.

Reply:
One idea is to subclass this interface with two types, like (demo not real code):

```
type StoreBatch struct {
        // key type is []byte
	deletes    map[[]byte]struct{}
        ...
}

type TableStoreBatch struct {
        // key type is TableKey
	deletes    map[TableKey]struct{} 
        ...
}
```

It establishes common interface but users doesn't need to use template. Wdyt?

Reply:
It's a good pattern to have a Base<T> and then subclass it with CaseA and CaseB without template. The interface helps constrain what they can be.

Reply:
done